test/integration/federation/server_test.go

```diff
@@ -69,11 +76,32 @@ func TestRun(t *testing.T) {
 	if err := waitForApiserverUp(); err != nil {
 		t.Fatalf("%v", err)
 	}
+}
+
+// Verifies that only default APIs are enabled when no runtime config is set.
+func TestDefaultRun(t *testing.T) {
+	run(t, "")
+	expectedGroupVersions := enabledGroupVersions
 	testSwaggerSpec(t)
 	testSupport(t)
-	testAPIGroupList(t)
-	testAPIGroup(t)
-	testAPIResourceList(t)
+	testAPIGroupList(t, expectedGroupVersions)
+	testAPIGroup(t, expectedGroupVersions)
+	testAPIResourceList(t, expectedGroupVersions)
+}
+
+// Verifies that all APIs are enabled when runtime config is set to all.
+func TestRunWithRuntimeConfigAll(t *testing.T) {
+	securePort = 6443 + 5
```

Comment:
I would like to get it in 1.6 if the release team agrees.
To clarify, this should not lead to flakes. Either the tests will always fail or pass.